cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -346,14 +374,396 @@ void AddSortingKernels(VectorKernel base, VectorFunction* func) {
   }
 }
 
+class TableSorter : public TypeVisitor {
+ private:
+  struct ResolvedSortKey {
+    ResolvedSortKey(const ChunkedArray& chunked_array, const SortOrder order)
+        : order(order) {
+      type = chunked_array.type().get();
+      null_count = chunked_array.null_count();
+      num_chunks = chunked_array.num_chunks();
+      for (const auto& chunk : chunked_array.chunks()) {
+        chunks.push_back(chunk.get());
+      }
+    }
+
+    template <typename ArrayType>
+    ArrayType* ResolveChunk(int64_t index, int64_t& chunk_index) const {
```

Comment:
Can you add a comment describing what this method does?

Reply:
Also, please make it `int64_t* chunk_index`. Out-parameters should be pointers.
(you could also return a `std::pair` or a dedicated struct... anyway)

Reply:
Added comments and changed to use new dedicated struct.